operator/src/network.rs

```diff
@@ -0,0 +1,99 @@
+use std::{sync::Arc, time::Duration};
+
+use kube::{
+    api::{Patch, PatchParams},
+    client::Client,
+    runtime::controller::Action,
+    Api, CustomResource, ResourceExt,
+};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+#[derive(CustomResource, Serialize, Deserialize, Debug, PartialEq, Clone, JsonSchema)]
+#[kube(
+    group = "keramik.3box.io",
+    version = "v1",
+    kind = "Network",
+    plural = "networks",
+    status = "NetworkStatus",
+    derive = "PartialEq",
+    namespaced
+)]
+pub struct NetworkSpec {
+    pub enabled: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, JsonSchema)]
+pub struct NetworkStatus {
+    enabled: bool,
+}
+
+//can add resource apis here
+pub struct ContextData {
+    client: Client,
+}
+
+impl ContextData {
+    pub fn new(client: Client) -> Self {
+        ContextData { client }
+    }
+}
+
+pub fn on_error(network: Arc<Network>, error: &Error, _context: Arc<ContextData>) -> Action {
+    eprintln!("Rec error:\n{:?}.\n{:?}", error, network);
+    Action::requeue(Duration::from_secs(5))
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Kube error: {source}")]
+    KubeError {
+        #[from]
+        source: kube::Error,
+    },
+}
+
+pub async fn reconcile(network: Arc<Network>, context: Arc<ContextData>) -> Result<Action, Error> {
+    let client: Client = context.client.clone();
+    let net_api: Api<Network> = Api::all(client);
+    let name = network.name_any();
+
+    let status = json!({
+        "status": NetworkStatus { enabled: true }
+    });
+
+    // TODO fails, assuming status/resource was not created
+    net_api
```

Comment:
noticed this was properly formatted/linted now, what should i be using locally during development?

Reply:
You can configure vscode to format using the rust analyzer on save. That is what I typically do (except with vim)

Alternatively you can run `cargo fmt` to format all files in the workspace.